src/wasm/literal.cpp

```diff
@@ -978,8 +978,14 @@ Literal Literal::neg() const {
 Literal Literal::abs() const {
   switch (type.getBasic()) {
     case Type::i32:
+      if (i32 == std::numeric_limits<int32_t>::min()) {
+        return *this;
+      }
       return Literal(std::abs(i32));
     case Type::i64:
+      if (i64 == std::numeric_limits<int64_t>::min()) {
+        return *this;
+      }
```

Comment:
Wait, wasm doesn't have integer `min`, does it? Unless this is for SIMD somehow?

Reply:
Oh, it's not a wasm instruction, it's some internal computation over literals... in that case we don't have a spec to compare against. Why is returning the (negative) literal the right value for `abs` here? (I would guess maxint)

Reply:
Out of curiosity, I found a testcase that uses this,
```wat
 (func $optimize-boolean (type $1) (param $0 i32) (param $1 i64) (result i32)
  (select
   (i32.const 0)
   (i32.const 0)
   (i32.rem_s
    (local.get $0)
    (i32.const 0)
   )
  )
 )
```
`--optimize-instructions` ends up doing `abs` on an integer here.

Reading the pass source, I'm not sure if this is a bug or not.

If you can confirm this PR doesn't change the outcome, that sounds ok, but I guess it's UB so we aren't sure it was consistent before?

Reply:
There is an integer abs SIMD instruction, and we run a spec test for its behavior that shows this is WAI.

Reply:
Ok, sgtm then. Meanwhile I verified all uses of integer `abs` in `OptimizeInstructions` are valid.